experiments/standalone/neural_snow/train_test_tools.jl

```diff
@@ -0,0 +1,351 @@
+using Flux
+using DataFrames, Dates, StatsBase
+using Flux.Data: DataLoader
+using Flux: loadparams!
+
+"""
+    show_mem_usage(ret)
+
+Print the current RAM usage (percentage).
+
+# Arguments
+- `ret::Bool`: indicates whether to return the percentage as a number.
+"""
+function show_mem_usage(ret::Bool = false)
+    used_mem = ((Sys.total_memory() / 2^20) - (Sys.free_memory() / 2^20)) / 1000.0
+    perc = 100.0 * (1.0 - Sys.free_memory()/Sys.total_memory())
+    print("RAM Usage: ", round(used_mem, digits = 3), " GB (", round(perc, digits = 2), "%)\n")
+    if ret
+        return used_mem
+    end
+end
+
+"""
+    make_data(data, input_vars, target, target_scale; testidx, dtype)
+
+Create training and testing matrices for the models from prepped input data.
+
+# Arguments
+- `data::DataFrame`: the data to be used for training and validation data.
+- `input_vars::Vector{Symbol}`: The set of features to be extracted as input data
+- `target::Symbol`: The feature to be extracted as the target variable
+- `target_scale::Real`: The scaling to apply to the output data
+- `testidx': the set of indices to use as test/validation data, or "nothing". Default is "nothing"
+- `dtype::Type`: The data type consistent with the model. Default is Float32.
+"""
+function make_data(data::DataFrame, input_vars::Vector{Symbol}, target::Symbol, target_scale::Real; testidx=nothing, dtype::Type=Float32)
```

Comment:
is it possible to use an existing train/test split function from MLUtils? Can we do a random split instead of passing in specific IDs?

e.g. MLUtils.splitobs(features, at=0.8)

(it's nice to rely on functionality in other established repos, since presumably their code has been tested and is maintained. less for us to maintain/test here).

They may also have some transformation functions (clipping, scaling) that we could make use of.